irc/handlers.go

```diff
@@ -2469,7 +2469,13 @@ func userHandler(server *Server, client *Client, msg ircmsg.IrcMessage, rb *Resp
 
 	err := client.SetNames(msg.Params[0], msg.Params[3], false)
 	if err == errInvalidUsername {
-		rb.Add(nil, server.name, ERR_INVALIDUSERNAME, client.t("Malformed username"))
+		// if client's using a unicode nick or something weird, let's just set 'em up with a blank username instead.
+		// fixes clients that just use their nick as a username so they can still use the interesting nick
+		if client.preregNick != "" && !isIdent(client.preregNick) {
```

Comment:
actually, should we check `client.preregNick == msg.Params[0]` to catch the exact case?

Reply:
hmmm, interesting. yeah we could definitely do something like that. I'm assuming we don't do any modification (casefolding/etc) of `client.preregNick`, if that's the case that should work fine too.

Reply:
Correct, we do a straight `client.preregNick = msg.Params[0]`

Reply:
perfect, awesome